update for latest template (#2363)
oneapi-src/oneAPI-samples: AI-and-Analytics/Getting-Started-Samples/INC-Sample-for-Tensorflow/README.md

> 

Author: yes, update

> This sample demonstrates using the Intel® Neural Compressor, which is part of the Intel® AI Tools with the with Intel® Optimizations for TensorFlow* to speed up inference by simplifying the process of converting the FP32 model to INT8/BF16.

Author: yes, update

You can achieve higher inference performance by converting the FP32 model to INT8 or BF16 model. Additionally, Intel® Deep Learning Boost (Intel® DL Boost) in Intel® Xeon® Scalable processors and Xeon® processors provides hardware acceleration for INT8 and BF16 models.

You will learn how to train a CNN model with Keras and TensorFlow*, use Intel® Neural Compressor to quantize the model, and compare the performance to see the benefit of Intel® Neural Compressor.

## Prerequisites

| Optimized for                     | Description |
|:---                               |:---         |
| OS                                | Ubuntu* 20.04 (or newer) <br> Windows 11, 10* |
| Hardware                          | Intel® Core™ Gen10 Processor <br> Intel® Xeon® Scalable Performance processors |
| Software                          | Intel® Neural Compressor, Intel Optimization for TensorFlow |

> 

Author: yes, update

|:---                               |:---         |
| OS                                | Ubuntu* 20.04 (or newer) <br> Windows 11, 10* |
| Hardware                          | Intel® Core™ Gen10 Processor <br> Intel® Xeon® Scalable Performance processors |
| Software                          | Intel® Neural Compressor, Intel® Extension for TensorFlow* |
> **Note**: AI and Analytics samples are validated on AI Tools Offline Installer. For the full list of validated platforms refer to [Platform Validation](https://github.com/oneapi-src/oneAPI-samples/tree/master?tab=readme-ov-file#platform-validation).

### Intel® Neural Compressor and Sample Code Versions

> 

Reviewer: It seems like Intel® AI Tools has been renamed to just AI Tools, we need to update this everywhere in the README

Author: yes, update

> 

Reviewer: Section 'For Local Development Environments' - we already cover all the needed instructions in Env Setup and Run a sample sections, so this section is redundant and should be removed

Author: yes, remove it.

- Test the performance of the FP32 model and INT8 (quantization) model.

## Environment Setup

You will need to download and install the following toolkits, tools, and components to use the sample.

If you have already set up the PIP or Conda environment and installed AI Tools go directly to Run the Notebook.

### 1. Get AI Tools

> Required AI Tools: Intel® Neural Compressor, Intel® Extension for TensorFlow* (GPU)

Reviewer: seems like this sample is a CPU sample?
```suggestion
Required AI Tools: Intel® Neural Compressor, Intel® Extension for TensorFlow* (CPU)
```

Author: yes, update

If you have not already, select and install these Tools via [AI Tools Selector](https://www.intel.com/content/www/us/en/developer/tools/oneapi/ai-tools-selector.html). AI and Analytics samples are validated on AI Tools Offline Installer. It is recommended to select Offline Installer option in AI Tools Selector.

> Note: If Docker option is chosen in AI Tools Selector, refer to [Working with Preset Containers](https://github.com/intel/ai-containers/tree/main/preset) to learn how to run the docker and samples.

Reviewer: fixing formatting for a note
```suggestion
>**Note**: If Docker option is chosen in AI Tools Selector, refer to [Working with Preset Containers](https://github.com/intel/ai-containers/tree/main/preset) to learn how to run the docker and samples.
```

Author: yes, update

### 2. (Offline Installer) Activate the AI Tools bundle base environment

If the default path is used during the installation of AI Tools:

```
source $HOME/intel/oneapi/intelpython/bin/activate
```

If a non-default path is used:

```
source <custom_path>/bin/activate
```

### 3. (Offline Installer) Activate relevant Conda environment

```
conda activate tensorflow
```

### 4. Clone the GitHub repository

```
git clone https://github.com/oneapi-src/oneAPI-samples.git
cd oneAPI-samples/AI-and-Analytics/Getting-Started-Samples/INC-Sample-for-Tensorflow
```

### 5. Install dependencies

Install for Jupyter Notebook:

```

> conda install jupyter runipy notebook -y

Reviewer: it's recommended to but dependent packages into requirements.txt file and install everything with `pip install -r requirements.txt`.
In any case, we need to make sure we list all the needed packages, right now I see that ipykernel and matplotlib packages are missing in the command

Author: yes, create requirements.txt

```

For Jupyter Notebook, refer to [Installing Jupyter](https://jupyter.org/install) for detailed installation instructions.

## Run the Sample

> **Note**: Before running the sample, make sure [Environment Setup](https://github.com/oneapi-src/oneAPI-samples/tree/master/AI-and-Analytics/Getting-Started-Samples/INC-Sample-for-TensorFlow#environment-setup) is completed.
>
Go to the section which corresponds to the installation method chosen in [AI Tools Selector](https://www.intel.com/content/www/us/en/developer/tools/oneapi/ai-tools-selector.html) to see relevant instructions:

* [AI Tools Offline Installer (Validated)](#ai-tools-offline-installer-validated)
* [Conda/PIP](#condapip)
* [Docker](#docker)

### AI Tools Offline Installer (Validated)

#### 1. Register Conda kernel to Jupyter Notebook kernel

> * Note: If you have done this step before, skip it.

Reviewer: note formatting
```suggestion
> **Note**: If you have done this step before, skip it.
```

Author: yes, update

If the default path is used during the installation of AI Tools:

```
$HOME/intel/oneapi/intelpython/envs/<offline-conda-env-name>/bin/python -m ipykernel install --user --name=tensorflow
```

If a non-default path is used:

```
<custom_path>/bin/python -m ipykernel install --user --name=tensorflow
```

#### 2. Launch Jupyter Notebook

> - Alternative 1: Launch Jupyter Notebook.

Reviewer: ```suggestion
- Option A: Launch Jupyter Notebook.
```

Author: yes, update

   ```
   jupyter notebook --ip=0.0.0.0
   ```

> - Alternative 2: You can launch Jupyter Notebook by running the script located in the sample code directory.

Reviewer: ```suggestion
- Option B: You can launch Jupyter Notebook by running the script located in the sample code directory.
```

	```
	(tensorflow) xxx@yyy:$ [I 09:48:12.622 NotebookApp] Serving notebooks from local directory:
	...
	[I 09:48:12.622 NotebookApp] Jupyter Notebook 6.1.4 is running at:
	[I 09:48:12.622 NotebookApp] http://yyy:8888/?token=146761d9317552c43e0d6b8b6b9e1108053d465f6ca32fca
	[I 09:48:12.622 NotebookApp]  or http://127.0.0.1:8888/?token=146761d9317552c43e0d6b8b6b9e1108053d465f6ca32fca
	[I 09:48:12.622 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
	[C 09:48:12.625 NotebookApp]

		To access the notebook, open this file in a browser:
			...
		Or copy and paste one of these URLs:
			http://yyy:8888/?token=146761d9317552c43e0d6b8b6b9e1108053d465f6ca32fca
		 or http://127.0.0.1:8888/?token=146761d9317552c43e0d6b8b6b9e1108053d465f6ca32fca
	[I 09:48:26.128 NotebookApp] Kernel started: bc5b0e60-058b-4a4f-8bad-3f587fc080fd, name: python3
	[IPKernelApp] ERROR | No such comm target registered: jupyter.widget.version

	```

#### 3. Follow the instructions to open the URL with the token in your browser

#### 4. Select the Notebook

```
inc_sample_tensorflow.ipynb
```

> #### 5. Change the kernel to **tensorflow**

Reviewer: fixing formatting
```suggestion
#### 5. Change the kernel to `tensorflow`
```

Author: yes, update

#### 6. Run every cell in the Notebook in sequence

### Conda/PIP

> - Note: Before running the instructions below, make sure your Conda/Python environment with AI Tools installed is activated

Reviewer: ```suggestion
> **Note**: Before running the instructions below, make sure your Conda/Python environment with AI Tools installed is activated
```

Author: yes, update

#### 1. Register Conda/Python kernel to Jupyter Notebook kernel

> * Note: If you have done this step before, skip it.

Reviewer: ```suggestion
> **Note**: If you have done this step before, skip it.
```

Author: yes, update

For Conda:
```
<CONDA_PATH_TO_ENV>/bin/python -m ipykernel install --user --name=tensorflow
```

To know <CONDA_PATH_TO_ENV>, run conda env list and find your Conda environment path.

For PIP:

```
python -m ipykernel install --user --name=tensorflow
```

#### 2. Launch Jupyter Notebook

> - Alternative 1: Launch Jupyter Notebook.

Reviewer: ```suggestion
- Option A: Launch Jupyter Notebook.
```

Author: yes, update

   ```
   jupyter notebook --ip=0.0.0.0
   ```

> - Alternative 2: You can launch Jupyter Notebook by running the script located in the sample code directory.

Reviewer: ```suggestion
- Option B: You can launch Jupyter Notebook by running the script located in the sample code directory.
```

Author: yes, update

## Environment Setup
If you have already set up the PIP or Conda environment and installed AI Tools go directly to Run the Notebook.

> 

Author: I will update the chapter structure for above style.
The requirements.txt is used by PIP.
But in this sample, we use conda. So no need to create and use requirements.txt file.

> ### On Linux*

Author: yes, update

> > Linux*:

Author: yes, update

Go to the section which corresponds to the installation method chosen in [AI Tools Selector](https://www.intel.com/content/www/us/en/developer/tools/oneapi/ai-tools-selector.html) to see relevant instructions:
* [AI Tools Offline Installer (Validated)](#ai-tools-offline-installer-validated)

> * [Conda/PIP](#setup-conda-environment)

Author: yes, add them.

### AI Tools Offline Installer (Validated)

> 1. If you have not already done so, activate the AI Tools bundle base environment.

Reviewer: Suggested content for AI Tools Offline Installer section:
 
**1. Register Conda kernel to Jupyter Notebook kernel**

If the default path is used during the installation of AI Tools:
```
$HOME/intel/oneapi/intelpython/envs/tensorflow-gpu/bin/python -m ipykernel install --user --name=tensorflow-gpu
```
If a non-default path is used:
```
<custom_path>/bin/python -m ipykernel install --user --name=tensorflow-gpu
```
**2. Launch Jupyter Notebook** 
<!-- add other flags to jupyter notebook command if needed, such as port 8888 or allow-root -->
```
jupyter notebook --ip=0.0.0.0
```
**3. Follow the instructions to open the URL with the token in your browser**

**4. Select the Notebook**
<!-- add sample file name -->
```
inc_sample_tensorflow.ipynb
```
**5. Change the kernel to `tensorflow-gpu`**
  <!-- specify relevant kernel name(s), for example `pytorch` -->
**6. Run every cell in the Notebook in sequence**

Author: yes, update

   ```
2. Change to the sample directory.

### Launch Jupyter Notebook

1. Launch Jupyter Notebook.
   ```

>    jupyter notebook --ip=0.0.0.0

Reviewer: Suggested content for Conda/PIP section:
 
### Conda/PIP
> **Note**: Before running the instructions below, make sure your Conda/Python environment with AI Tools installed is activated

**1. Register Conda/Python kernel to Jupyter Notebook kernel** 
<!-- keep placeholders in this step, user could use any name for Conda/PIP env -->
For Conda:
```
<CONDA_PATH_TO_ENV>/bin/python -m ipykernel install --user --name=<your-env-name>
```
To know <CONDA_PATH_TO_ENV>, run `conda env list` and find your Conda environment path.

For PIP:
```
python -m ipykernel install --user --name=<your-env-name>
```
**2. Launch Jupyter Notebook**
<!-- add other flags to jupyter notebook command if needed, such as port 8888 or allow-root --> 
```
jupyter notebook --ip=0.0.0.0
```
**3. Follow the instructions to open the URL with the token in your browser**

**4. Select the Notebook**
<!-- add sample file name -->
```
inc_sample_tensorflow.ipynb
```
**5. Change the kernel to `<your-env-name>`**
<!-- leave <your-env-name> as a placeholder as user could choose any name for the env -->

**6. Run every cell in the Notebook in sequence**

Reviewer: Suggested content for Docker section:

### Docker
AI Tools Docker images already have Get Started samples pre-installed. Refer to [Working with Preset Containers](https://github.com/intel/ai-containers/tree/main/preset) to learn how to run the docker and samples.

	[IPKernelApp] ERROR | No such comm target registered: jupyter.widget.version

	```

#### 3. Follow the instructions to open the URL with the token in your browser

#### 4. Select the Notebook

> inc_sample_tensorflow.ipynb

Reviewer: formatting
```
inc_sample_tensorflow.ipynb
```

Author: yes, update

> #### 5. Change the kernel to **tensorflow**

Reviewer: ```suggestion
#### 5. Change the kernel to `tensorflow`
```

Author: yes, update
